### PR TITLE
remove default.qubit.autograd

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-72795]

### CODE REVIEW

File: tests/devices/test_default_qubit_legacy.py [hunk 1]
```diff
@@ -1006,9 +1006,7 @@ def test_defines_correct_capabilities(self):
             "supports_inverse_operations": True,
             "supports_analytic_computation": True,
             "supports_broadcasting": True,
-            "passthru_devices": {
-                "autograd": "default.qubit.autograd",
-            },
+            "passthru_devices": {},
```
Comment (astralcai) [inline tests/devices/test_default_qubit_legacy.py:1009]: What is `passthru_devices`, and is this even needed anymore?

**Review by astralcai — APPROVED** [review]
Love to see it. ❤️

**Review by PietropaoloFrisoni — COMMENTED** [review]
Is there any specific reason why we have not updated `deprecations.rst` for devices?

**Comment by albi3ro** [comment]
> Is there any specific reason why we have not updated `deprecations.rst` for devices?

Good point.  Since the entry covered five PR's, I was putting it off till the last one, but I can do so now so we don't forget.

File: doc/development/deprecations.rst [hunk 2]
```diff
@@ -88,6 +82,12 @@ Other deprecations
 Completed deprecation cycles
 ----------------------------
 
+* All of the legacy devices (any with the name ``default.qubit.{autograd,torch,tf,jax,legacy}``) are deprecated. Use ``default.qubit`` instead,
```
Comment (PietropaoloFrisoni) [inline doc/development/deprecations.rst:85]:
```suggestion
* All of the legacy devices (any with the name ``default.qubit.{autograd,torch,tf,jax,legacy}``) are removed. Use ``default.qubit`` instead,
```

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks! Hopefully, this will make things easier and faster : )

**Pull request merged**